vochain/scrutinizer/process.go

```diff
@@ -6,21 +6,65 @@ import (
 	"strings"
 	"time"
 
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/timshannon/badgerhold/v3"
 	"go.vocdoni.io/proto/build/go/models"
+	"google.golang.org/protobuf/proto"
 
 	"go.vocdoni.io/dvote/log"
 	"go.vocdoni.io/dvote/types"
+	scrutinizerdb "go.vocdoni.io/dvote/vochain/scrutinizer/db"
 	"go.vocdoni.io/dvote/vochain/scrutinizer/indexertypes"
 )
 
+var zeroBytes = []byte("")
+
+func nonNullBytes(p []byte) []byte {
+	if p == nil {
+		return zeroBytes
+	}
+	return p
+}
+
+// TODO(mvdan): funcs to safely convert integers
+
+func encodedPb(msg proto.Message) types.EncodedProtoBuf {
+	enc, err := proto.Marshal(msg)
+	if err != nil {
+		panic(err) // TODO(mvdan): propagate errors via a database/sql interface?
+	}
+	return nonNullBytes(enc)
+}
+
 // ProcessInfo returns the available information regarding an election process id
 func (s *Scrutinizer) ProcessInfo(pid []byte) (*indexertypes.Process, error) {
-	startTime := time.Now()
-	defer func() { log.Debugf("ProcessInfo took %s", time.Since(startTime)) }()
+	bhStartTime := time.Now()
 	proc := &indexertypes.Process{}
-	err := s.db.FindOne(proc, badgerhold.Where(badgerhold.Key).Eq(pid))
-	return proc, err
+	if err := s.db.FindOne(proc, badgerhold.Where(badgerhold.Key).Eq(pid)); err != nil {
+		return nil, err
+	}
+	log.Debugf("ProcessInfo badgerhold took %s", time.Since(bhStartTime))
+
+	sqlStartTime := time.Now()
+
+	queries, ctx, cancel := s.timeoutQueries()
+	defer cancel()
+	sqlProcInner, err := queries.GetProcess(ctx, pid)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("ProcessInfo sqlite took %s", time.Since(sqlStartTime))
+	sqlProc := indexertypes.ProcessFromDB(&sqlProcInner)
```

Comment:
not sure I follow; I need to convert from the sql type to our custom indexertypes type. that requires extra code to run, which I put in ProcessFromDB. I can't make GetProcess do it directly because it's code generated by sqlc.

Reply:
Ah, I see. So, would it make sense to build an intermediate method to do both queries at once so abstract the developer from the SQL specific stuff?

Reply:
Ok. I'm merging that as a first stone for this refactor. Then let's see how it evolves.